src/main/java/de/srendi/advancedperipherals/common/addons/refinedstorage/RSCraftJob.java

```diff
@@ -0,0 +1,251 @@
+package de.srendi.advancedperipherals.common.addons.refinedstorage;
+
+import com.refinedmods.refinedstorage.api.autocrafting.preview.Preview;
+import com.refinedmods.refinedstorage.api.autocrafting.preview.PreviewType;
+import com.refinedmods.refinedstorage.api.autocrafting.status.TaskStatus;
+import com.refinedmods.refinedstorage.api.autocrafting.task.TaskId;
+import com.refinedmods.refinedstorage.api.network.autocrafting.AutocraftingNetworkComponent;
+import com.refinedmods.refinedstorage.api.resource.ResourceKey;
+import com.refinedmods.refinedstorage.api.storage.Actor;
+import dan200.computercraft.api.peripheral.IComputerAccess;
+import de.srendi.advancedperipherals.AdvancedPeripherals;
+import de.srendi.advancedperipherals.common.util.StatusConstants;
+import de.srendi.advancedperipherals.common.util.inventory.BasicCraftJob;
+import net.minecraft.world.level.Level;
+
+import java.util.Collections;
+import java.util.Optional;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.stream.Collectors;
+
+public class RSCraftJob extends BasicCraftJob {
+
+    private final AutocraftingNetworkComponent autocraftingComponent;
+    private final ResourceKey toCraft;
+
+    private TaskStatus craftingTask;
+    private Preview preview;
+    private Future<Optional<Preview>> futureCalculationResult;
+    private Future<Optional<TaskId>> futureTask;
+
+    public RSCraftJob(IComputerAccess computer, Level world, long amount, ResourceKey toCraft, AutocraftingNetworkComponent calculationResult) {
+        super(computer, "rs", world, amount);
+        this.autocraftingComponent = calculationResult;
+        this.toCraft = toCraft;
+    }
+
+    @Override
+    protected boolean isJobDone() {
+        return craftingTask != null && craftingTask.percentageCompleted() == 100;
```

Comment:
idk if `percentageCompleted` returns an int or a double, but if it returns a double I'd recommend compare with `>= 100` and `< 100`, since float point numbers sometimes can have weird precise problems.